HW03.Rmd

```diff
@@ -0,0 +1,144 @@
+  * Incorrect guess (e.g. the participant never guessed correctly despite it being calcuable)
+  * Unrewarding interventions (e.g. the participant chose to make LESS money when given a clear choice)
+  
+Just like with the timing scores above, I calculated each participant's scores for these factors (e.g. calculated number of incorrect guesses over total number of trials) and then Z-scored them.
+
+Here's that visualization (much like the one above!):
+
+*(Note: I split the directional bias plots into the middle two blocks (guess and intervene) and outer two blocks (observe and observe-repeat) because they are mechanically different from each other)*
+
+
+```{r Other Acceptability Scores}
+#Read in our CSV of data (pre-cleaned... in MATLAB)
+data_scoring <- read.csv("2lot_data_scoring.csv", stringsAsFactors = FALSE)
+
+#Pre-allocate color choices for aesthetic flexibility
+sd_color <- "orange" #color of standard deviation line
+mean_color <- "green" #color of mean line
+fill_color <- "dodgerblue3"
+mean_linetype <- "dashed" #linetype of mean line
+sd_linetype <- "dashed" #linetype of standard deviation line
+
+#Put data into lists so that each plot can be made in a loop
+score_list <- list()
+data_list <- list(data_scoring$left_obs_rep, data_scoring$left_gue_int, data_scoring$incorrect_guess, data_scoring$low_intervene)
+title_list <- list("Observe Directional Bias", "Middle Directional Bias", "Incorrect Guess", "Low Intervene")
+x_label_list <- list("% Bias", "% Bias", "%Incorrect Guess", "% Low Choice")
+
+#Create plots
+for(i in 1:length(data_list)){ #in a loop from 1 to the number of blocks, go through each block's data
+  score_list[[i]] <- local({ #save ggplot to this list
+    i <- i #make i local to limit variable scope within for loop
+    ggplot(data_scoring, aes(data_list[[i]])) + #plot current block's scores
+    geom_histogram(bins = 5, fill = fill_color) + #make histogram
+    labs(x = x_label_list[[i]], y = "Count", title = title_list[[i]], subtitle = "n = 25") + #labels
+    theme(plot.title = element_text(hjust = 0.5)) + #center title
+    geom_vline(xintercept = mean(data_list[[i]]), linetype = mean_linetype, color = mean_color) + #mean
+    geom_vline(xintercept = sd(data_list[[i]]), linetype = sd_linetype, color = sd_color) + #sd line
+    geom_vline(xintercept = -sd(data_list[[i]]), linetype = sd_linetype, color = sd_color)}) #sd line
+}
+
+#Plot the plots in a 2x2 grid!
+grid.arrange(score_list[[1]], score_list[[2]], score_list[[3]], score_list[[4]], nrow = 2)
+
+
+```
+
+
+## Participant Choice
+
+I've been using the grid.arrange() function for the above because I was dealing with subplots of completely different data variables that I wanted displayed together because they have a similar purpose. I would, however, like to have the opportunity to play around with facets, so now I'm going to load in actual data from Subject 7 of this study.
+
```

Comment:
I am very new to ggplot (before only used base R) and did not know about about grid.arrange, only facets. New thing I learned :)